tools/c7n_azure/c7n_azure/resources/public_ip.py

```diff
@@ -33,3 +34,12 @@ class resource_type(object):
             'properties.publicIPAllocationMethod',
             'properties.ipAddress'
         )
+
+    def get_resources(self, resource_ids):
```

Comment:
I believe it is actually possible to write this method generically for all of Azure using the provider list from the Azure API.  We use this already to lookup api versions for resources based on their ID and you'll see as part of doing that we look up the provider information which may have all you need to build a client to do a get (equivalent of our enum specs).  Not really sure, we can investigate later as we try to support more resource types for this sort of filter.